windows-containers/productization.md

```diff
@@ -0,0 +1,192 @@
+(kubelet, OVN, and the Windows Machine Config Bootstrapper) on user-provided
+Windows machines
+* Perform all the required steps within the node for it to be an OpenShift
+compute node
+* Administrator initiated upgrade of all OpenShift related components
+(kubelet, OVN and Windows Machine Config Bootstrapper) in the node
+
+### Non-Goals
+
+As part of this enhancement we do not plan to support:
+* Windows node provisioning or de-provisioning
+* Installing the container runtime in the Windows node
+* Windows operating system upgrades
+* Node management like reboots and node draining
+* OpenShift Builds
+* Network configuration
+  * The details of this will be worked out as part of the enhancement
+    proposal for "OVN Plumbing for Hybrid Linux+Windows Clusters GA"
+
+## Proposal
+
+The two main pieces that will allow us to achieve our goals is a Windows Scale
+Up (WSU) Ansible playbook and a Windows Machine Config Bootstrapper (WMCB)
+binary. The Ansible playbook will collect information from the cluster and
+transfer it to the Windows node. The WMCB will then perform the necessary
+steps on the node for it to join the cluster.
+
+### Justification
+
+The reason for having an Ansible playbook and on-node executable split is to
+have a consistent user experience with "Bring your own RHEL". It also allows
+us to be future proof when we move to an operator workflow. Please read the
+[alternatives](#Alternatives) section.
+
+The reason we are using a binary rather than a container image is
+that all Windows container images contains a Windows kernel and Red Hat has a
+policy to not ship 3rd party kernels for support reasons.
+
+### Implementation Details
+
+#### Windows Scale Up (WSU)
+
+The Windows Scale Up is an Ansible playbook that has the follow prerequisites:
+* Needs to run on a Linux system
+* Needs to be able to access the cluster where the Windows compute will be added
+* Needs to be able to access the Windows node
+
+The inputs that the WSU playbook requires are:
+* Windows node credentials
+* Kubelet download location
+* Worker Ignition
+* WMCB download location
+
+The actions that the WSU will perform are:
+* Check if the container runtime is present on the Windows node
+* Download and copy the kubelet to the Windows node
+* Extract the worker Ignition from the cluster and copy it to the Windows node
+* Download the WMCB, copy it to the Windows node and execute it
+
+#### Windows Machine Config Bootstrapper (WMCB)
+
+The Windows Machine Config Bootstrapper executable has the following
+prerequisites:
+* Only supports Windows x86-64 nodes (Windows Server 2019)
+* Can only run on Windows nodes
+
+The inputs that the WMCB requires are:
+* kubelet location on the local disk
+* Worker Ignition location on the local disk
+
+The actions that the WMCB will perform are:
+* Install / upgrade and configure the kubelet
+* Parse the worker Ignition and extract the bootstrap kubeconfig and the
```

Comment:
kubelet configuration varies for linux versus windows hosts, this has to be a separate pool.

Reply:
It's only used for the cluster coordinates and bootstrapping credentials.